File: KB boot/week1/완주하지 못한 선수.java
import java.util.HashMap;

class Solution {
    public String solution(String[] participant, String[] completion) {
        HashMap<String, Integer> map = new HashMap();
        
        for (String player : participant) {
            if (map.containsKey(player)) {
                map.put(player, map.get(player) + 1);
            } else {
                map.put(player, 1);
            }
        }
        
        for (String player : completion) {
            map.put(player, map.get(player) - 1);
        }
        
        for (String player : map.keySet()) {
            if (map.get(player) != 0) {
                return player;
            }
        }
        
        return "";
    }
}

File: Programmers/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    string answer = "";
    unordered_map<string, int> s;
    
    for (const auto& name : participant) {
        s[name]++;
    }
    
    for (const auto& name : completion) {
        s[name]--;
    }    
    
    for (const auto& [name, c] : s) {
        if (c > 0) {
            return name;
        }
    }
    return "";
}
